app/controllers/search_controller.rb

```diff
@@ -1,5 +1,5 @@
 class SearchController < ApplicationController
   def new
-    @results = Resource.search(params[:query])
+    @results = params[:query].present? ? Resource.search(params[:query]) : []
```

Comment:
But these are minor things, this should merge regardless. Null objects can come later.